src/System.Runtime.Analyzers/Core/SpecifyIFormatProvider.cs

```diff
@@ -27,42 +34,203 @@ public abstract class SpecifyIFormatProviderAnalyzer : DiagnosticAnalyzer
+                         targetMethod.Equals(stringFormatMemberWithStringObjectObjectAndObjectParameter) ||
+                         targetMethod.Equals(stringFormatMemberWithStringAndParamsObjectParameter)))
+                    {
+                        // Sample message for IFormatProviderAlternateStringRule: Because the behavior of string.Format(string, object) could vary based on the current user's locale settings,
+                        // replace this call in IFormatProviderStringTest.M() with a call to string.Format(System.IFormatProvider, string, params object[]). If the result of 
+                        // string.Format(System.IFormatProvider, string, params object[]) will be displayed to the user, specify System.Globalization.CultureInfo.CurrentCulture as the 'IFormatProvider' parameter.
+                        // Otherwise, if the result will be stored and accessed by software, such as when it is persisted to disk or to a database, specify System.Globalization.CultureInfo.InvariantCulture.
+                        oaContext.ReportDiagnostic(
+                            invocationExpression.Syntax.CreateDiagnostic(
+                                IFormatProviderAlternateStringRule,
+                                targetMethod.ToDisplayString(SymbolDisplayFormat.CSharpErrorMessageFormat),
+                                oaContext.ContainingSymbol.ToDisplayString(SymbolDisplayFormat.CSharpErrorMessageFormat),
+                                stringFormatMemberWithIFormatProviderStringAndParamsObjectParameter.ToDisplayString(SymbolDisplayFormat.CSharpErrorMessageFormat),
+                                currentCultureProperty.ToDisplayString(SymbolDisplayFormat.CSharpErrorMessageFormat),
+                                invariantCultureProperty.ToDisplayString(SymbolDisplayFormat.CSharpErrorMessageFormat)));
+
+                        return;
+                    }
+                    #endregion
+
+                    #region "IFormatProviderAlternateStringRule & IFormatProviderAlternateRule"
+                    IEnumerable<IMethodSymbol> methodsWithSameNameAsTargetMethod = targetMethod.ContainingType.GetMembers(targetMethod.Name).OfType<IMethodSymbol>();
+                    if (methodsWithSameNameAsTargetMethod.Count() > 1)
+                    {
+                        var correctOverloads = methodsWithSameNameAsTargetMethod.GetMethodOverloadsWithDesiredParameterAtLeadingOrTrailing(targetMethod, IFormatProviderType);
+                        
+                        // If there are two matching overloads, one with CultureInfo as the first parameter and one with CultureInfo as the last parameter,
+                        // report the diagnostic on the overload with CultureInfo as the last parameter, to match the behavior of FxCop.
+                        var correctOverload = correctOverloads
+                                              .Where(overload => overload.Parameters.Last().Type.Equals(IFormatProviderType))
```

Comment:
One of the checks in `GetMethodOverloadsWithDesiredParameterAtLeadingOrTrailing` is to make sure the overload exactly +1 no of parameters than the target method. So this can never be empty.